app/client/src/components/shared/MapWidgets.tsx

```diff
@@ -1378,6 +1504,215 @@ function ShowAllWaterbodies({
   );
 }
 
+type ShowSurroundingMonitoringLocationsProps = {
+  getDisabled: () => boolean;
+  getMonitoringLocations: () => FetchState<MonitoringLocationsData>;
+  mapView: __esri.MapView;
+  services: ServicesState;
+  setDisabled: (disabled: boolean) => void;
+  setVisible: (visible: boolean) => void;
+  surroundingMonitoringLocationsLayer: __esri.FeatureLayer | '';
+};
+
+// Defines the show all waterbodies widget
+function ShowSurroundingMonitoringLocations({
+  getDisabled,
+  getMonitoringLocations,
+  mapView,
+  services,
+  setDisabled,
+  setVisible,
+  surroundingMonitoringLocationsLayer,
+}: ShowSurroundingMonitoringLocationsProps) {
+  const [hover, setHover] = useState(false);
+
+  const [surroundingMonitoringLocations, setSurroundingMonitoringLocations] =
+    useState<FetchState<MonitoringLocationsData>>({
+      status: 'fetching',
+      data: null,
+    });
+
+  const [watcherInitialized, setWatcherInitialized] = useState(false);
+  useEffect(() => {
+    if (!surroundingMonitoringLocationsLayer || watcherInitialized) return;
+    if (services.status !== 'success') return;
+
+    type SimpleExtent = {
+      xmin: number;
+      xmax: number;
+      ymin: number;
+      ymax: number;
+    };
+    let lastExtent: SimpleExtent | null = null;
+
+    let abortController = new AbortController();
+
+    // watch for extent changes and query the waterquality portal
+    reactiveUtils.watch(
+      () => mapView.stationary,
+      () => {
+        if (!mapView.stationary) return;
+        if (!mapView.ready) return;
+
+        const newExtent: SimpleExtent = {
+          xmin: mapView.extent.xmin,
+          xmax: mapView.extent.xmax,
+          ymin: mapView.extent.ymin,
+          ymax: mapView.extent.ymax,
+        };
+        if (JSON.stringify(newExtent) === JSON.stringify(lastExtent)) return;
+        lastExtent = newExtent;
+        abortController.abort();
+
+        setSurroundingMonitoringLocations({
+          status: 'fetching',
+          data: null,
+        });
+        abortController = new AbortController();
+
+        // convert the extent into northwest and southeast corner points
+        const northwestWM = new Point({
+          x: mapView.extent.xmin,
+          y: mapView.extent.ymax,
+        });
+        const southwestWM = new Point({
```

Comment:
This is just a variable name, but it could be confusing in the future:
```suggestion
        const southeastWM = new Point({
```